view/src/storage.rs

```diff
@@ -117,9 +122,49 @@ impl Storage {
         Ok(Storage {
             pool,
             uncommitted_height: Arc::new(Mutex::new(None)),
+            scanned_notes_tx: broadcast::channel(10).0,
         })
     }
 
+    pub fn await_change(
+        &self,
+        note_commitment: tct::Commitment,
+    ) -> impl Future<Output = anyhow::Result<NoteRecord>> + '_ {
+        let mut rx = self.scanned_notes_tx.clone().subscribe();
```

Comment:
I think the clone here isn't needed, since subscribe() will create an owned receiver, so it's already unlinked from the lifetime of the sender.